### PR TITLE
Feature/movie details

### PR DESCRIPTION
**Pull request by khaledsaadeh**
Created the movie details page, and updated its route paths.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -12,26 +14,31 @@ const nowPlayingAPI =
...
-        setMovies(res.data.results);
-      });
+    axios.get(nowPlayingAPI).then((res) => {
+      setMovies(res.data.results);
+    });
   }, []);
 
   const renderSearchResults = (movies) => {
     setMovies(movies);
   };
```
Comment (MuhammadObeidat) [inline src/App.js:26]: It's preferred to have this logic inside a movie component then import the whole component inside the app

File: src/App.js [hunk 2]
```diff
@@ -12,26 +14,31 @@ const nowPlayingAPI =
...
+          renderSearchResults={renderSearchResults}
+        />
+        <Route path="/" exact>
+          <MovieList movies={movies} />
+        </Route>
+        <Route path="/movies/:movieID">
+          <MovieDetails />
+        </Route>
+      </div>
+    </Switch>
```
Comment (MuhammadObeidat) [inline src/App.js:41]: It's better to have this inside a router component then use it here

File: src/components/MovieDetails.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      });
+  };
+
+  const fetchCast = () => {
+    axios
+      .get(
+        `https://api.themoviedb.org/3/movie/${params.movieID}/credits?api_key=4f298a53e552283bee957836a529baec&language=en-US`
+      )
+      .then((res) => {
+        console.log(res.data);
```
Comment (MuhammadObeidat) [inline src/components/MovieDetails.js:29]: Please always keep in mind to clear console log after coding

File: src/components/MovieList.js [hunk 4]
```diff
@@ -1,27 +1,34 @@
...
+import MovieDetails from "./MovieDetails";
+import { Link } from "react-router-dom";
 const MovieList = (props) => {
   const setColor = (rating) => {
-    
     if (rating >= 8) return "bg-green-400";
-    else if (rating >= 7 ) return "bg-yellow-400";
+    else if (rating >= 7) return "bg-yellow-400";
     else if (rating >= 6) return "bg-orange-400";
     else return "bg-red-400";
```
Comment (MuhammadObeidat) [inline src/components/MovieList.js:9]: Could please use switch statement here, it shoulld be something like this

switch(rating) {
case 6: 
return "bg-orange-400";
case 7: 
return "bg-yellow-400";
case 8: 
return "bg-green-400";
default:
return "bg-red-400";
}

File: src/components/MovieList.js [hunk 5]
```diff
@@ -1,27 +1,34 @@
...
-            <h2 className="font-bold text-xl mb-2 m-4">{movie.title}</h2>
+            <Link to={`movies/${movie.id}`}>
+              <h2 className="font-bold text-xl mb-2 m-4">{movie.title}</h2>
+            </Link>
             <h3 className="m-4"> Release Date: {movie.release_date}</h3>
             <span
-              className={`${setColor(movie.vote_average)} font-bold rounded-xs p-1`}
+              className={`${setColor(
+                movie.vote_average
+              )} font-bold rounded-xs p-1`}
```
Comment (MuhammadObeidat) [inline src/components/MovieList.js:31]: It's totally right what did here, but we use a helper library clalled "classnames"
you can check it out for more details

**Comment by MuhammadObeidat** [comment]
It's looks perfect over all.

File: src/index.js [hunk 6]
```diff
@@ -2,12 +2,15 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
+import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';
 import reportWebVitals from './reportWebVitals';
 
 ReactDOM.render(
+  <BrowserRouter>
```
Comment (hazemturki) [inline src/index.js:9]: Format this page.

File: src/index.css [hunk 7]
```diff
@@ -1,3 +1,15 @@
...
 @tailwind utilities;
+
+@layer components {
+
+    .container-primary{
+        @apply  flex px-6 py-10  shadow-xl rounded bg-gradient-to-r from-gray-200 via-gray-400 to-gray-600
+    }
+
+    .info-card{
+      @apply  flex-shrink-0 shadow-xl m-2 w-32 h-56 max-w-xs overflow-hidden rounded-lg transform transition hover:scale-105
```
Comment (hazemturki) [inline src/index.css:12]: Let's revert this change.
Take these classes back to the component itself.

File: src/components/MovieList.js (hunk 4 above)
Comment (khaledsaadeh) [reply src/components/MovieList.js:9]: Since the rating is a float, I wasn't able to get the appropriate color through a switch statement, I had to use comparisons, otherwise it would always return "bg-red-400"

**Review by hazemturki — APPROVED** [review]
